src/node-jspm-jasmine.js

```diff
@@ -141,17 +180,73 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// the "onComplete" hook to prevent jasmine's self-righteous exit
+		jasmine.onComplete(function (passed) {
+			// avoid misalignment with jasmine's output
+			console.log('');
+			// import the rest of the modules not already imported (evaluated)
+			// as dependencies of specs
+			Promise.all(Object.keys(coverage.coverageFiles).map(function (file) {
+				return SystemJS.import(path.join(process.cwd(), file));
+			})).then(function () {
+				if (typeof __coverage__ === 'undefined') {
+					console.log("No coverage was collected for files matching globs " + coverage.files);
+				} else {
+					try {
+						const collector = remap(__coverage__)
+						let report = Report.create(
+							(coverage.reporter || 'html'),
+							{
+								dir: coverage.dir
+							}
+						);
+						report.writeReport(collector, true);
```

Comment:
Yeah that makes sense. I think that the source map situation with jspm might be related to an issue in babel, and that Guy Bedford has implemented a workaround in jspm but that it hasn't been released to jspm yet. See https://github.com/jspm/jspm-cli/issues/1804 for where I'm getting this.

Regarding whether we should be defensive or not with the source maps... I think that throwing a nice descriptive error is definitely better than dying in a hard-to-debug way. However, ideally I think we *would* try to adapt to the situation, just logging to the console that we cannot provide coverage for that particular file because jspm did not create a source map for that file. Do you see any problems with the defensive/no-error approach?